protocol/contact.go

```diff
@@ -363,14 +364,9 @@ func BuildContactFromPublicKey(publicKey *ecdsa.PublicKey) (*Contact, error) {
 	return buildContact(id, publicKey)
 }
 
-func getShortenedCompressedKey(publicKey string) string {
-	if len(publicKey) > 9 {
-		firstPart := publicKey[0:3]
-		ellipsis := "..."
-		publicKeySize := len(publicKey)
-		lastPart := publicKey[publicKeySize-6 : publicKeySize]
-		abbreviatedKey := fmt.Sprintf("%s%s%s", firstPart, ellipsis, lastPart)
-		return abbreviatedKey
+func getShortenedCompressedKey(compressedKey string) string {
+	if len(compressedKey) > 12 {
```

Comment:
This length check could be in the `ShortenedCompressedKey` function. The rationale is that callers of `ShortenedCompressedKey` will need to (ideally) always remember to check the length, otherwise there is risk of getting a panic with index out of bounds.

Reply:
Doesn't the correct check need to be
```go
	if len(compressedKey) != 12 {
		return ""
	}
	return alias.ShortenedCompressedKey(compressedKey)
```

Reply:
@saledjenic The `compressedKey` length is, as far as I've seen, longer than `12`, we just check for it being `12` because this is the minimum amount of chars to avoid an index out of bounds expection.

@ilmotta Totally makes sense! thank you

Reply:
@ulisesmac true, I wanted to say more about early return, but then out of any context added the wrong comparison. Sorry for confusion.